modules/rest-util/src/blaze/middleware/fhir/resource.clj

```diff
@@ -109,19 +132,43 @@
                     :http/status 415))
     (if (str/blank? (slurp (:body request)))
       (assoc request :body nil)
-      (ba/incorrect "Content-Type header expected, but is missing."))))
+      (ba/incorrect "Expected Content-Type header for FHIR resources."))))
+
+(defn- binary-resource-request [request]
+  (if-let [content-type (request/content-type request)]
+    (cond
+      (json-request? content-type) (resource-request-json request)
+      (xml-request? content-type) (resource-request-xml request)
+      :else
+      (resource-request-binary-data request))
+    (ba/incorrect "Expected Content-Type header for binary resources.")))
```

Comment:
my text from the resource-request function is: Content-Type header expected, but is missing.